python/doc/examples/meta_modeling/general_purpose_metamodels/plot_distribution_linear_regression.py

```diff
@@ -0,0 +1,250 @@
+"""
+Distribution of estimators in linear regression
+===============================================
+"""
+
+# %%
+# Introduction
+# ~~~~~~~~~~~~
+#
+# In this example, we are interested by the distribution of the estimator of the variance
+# of the observation error in linear regression.
+# We are also interested in the estimator of the standard deviation of the
+# observation error.
+# We show how to use the :class:`~openturns.PythonRandomVector` in order to
+# perform a study of the sample distribution of these estimators.
+#
+# In the general linear regression model, the observation error :math:`\epsilon` has the
+# Gaussian distribution :math:\cN(0, \sigma^2)` where :math:`\sigma > 0`
+# is the standard deviation.
+# We are interested in the estimators of the variance :math:`\sigma^2`
+# and the standard deviation :math:`\sigma`:
+#
+# - the variance :math:`\sigma^2` is estimated from :meth:`~openturns.LinearModelResult.getSigma2`;
+# - the standard deviation :math:`\sigma` is estimated from :meth:`~openturns.LinearModelAnalysis.getResidualsStandardError`.
+#
+# The asymptotic distribution of these estimators is known (see [vaart2000]_)
+# but we want to perform an empirical study, based on simulation.
+# In order to see the distribution of the estimator, we simulate an observation of the estimator,
+# and repeat that experiment :math:`r \in \Nset` times, where :math:`r`
+# is a large integer.
+# Then we approximate the distribution using :class:`~openturns.KernelSmoothing`.
+
+
+import openturns as ot
+import openturns.viewer as otv
+import pylab as pl
+
+# %%
+# The simulation engine
+# ~~~~~~~~~~~~~~~~~~~~~
+# The simulation is based on the :class`~openturns.PythonRandomVector` class,
+# which simulates independent observations of a random vector.
+# The `getRealization()` method implements the simulation of the observation
+# of the estimator.
+#
+
+
+class SampleEstimatorLinearRegression(ot.PythonRandomVector):
+    def __init__(
+        self, sample_size, true_standard_deviation, coefficients, estimator="variance"
+    ):
+        """
+        Create a RandomVector for an estimator from a linear regression model.
+
+        Parameters
+        ----------
+        sample_size : int
+            The sample size n.
+        true_standard_deviation : float
+            The standard deviation of the gaussian observation error.
+        coefficients: sequence of p floats
+            The coefficients of the linear model.
+        estimator: str
+            The estimator.
+            Available estimators are "variance" or "standard-deviation".
+
+        Returns
+        -------
+        None.
```

Comment:
you can drop the return section when there is notthin to return

Reply:
Ok. Fixed at 567b38bf6